services/QuillLMS/client/app/bundles/Teacher/components/assignment_flow/create_unit/assign_a_diagnostic.tsx

```diff
@@ -110,9 +162,66 @@ const FilterTab = ({ activeFilter, filter, setFilter, number, }) => {
 const AssignADiagnostic = ({ history, assignedPreTests, }) => {
   document.title = getPageTitle("Assign", "Diagnostic")
   const [filter, setFilter] = React.useState(ALL)
+  const [classCount, setClassCount] = React.useState<number>(null)
+  const [recommendationsCount, setRecommendationsCount] = React.useState<number>(null)
+  const [recommendedDiagnosticsByLevel, setRecommendedDiagnosticsByLevel] = React.useState<RecommendedDiagnosticsByLevelInterface>({})
+
+  React.useEffect(() => {
+    getCompleteHistory()
+  }, [])
+
+  function getCompleteHistory() {
+    requestGet('/diagnostic_insights/complete_history', (body) => {
+      let classCount = 0
+      let recommendationsCount = 0
+      const newRecommendedDiagnosticsByLevel = {
+        [STARTER]: [],
+        [INTERMEDIATE]: [],
+        [ADVANCED]: [],
+        [ELL_STARTER]: [],
+        [ELL_INTERMEDIATE]: [],
+        [ELL_ADVANCED]: []
+      }
+
+      Object.keys(body).forEach((classroomId) => {
+        const classroom = body[classroomId]
+        const ellLearningSequences = classroom.ell.filter(student => Object.keys(student.student_learning_sequences).length)
+        const nonELLLearningSequences = classroom.non_ell.filter(student => Object.keys(student.student_learning_sequences).length)
+
+        const ellRecommendations = generateRecommendations(true, ellLearningSequences, classroomId)
+        const nonELLRecommendations = generateRecommendations(false, nonELLLearningSequences, classroomId)
+
+        let ellRecommendationsCount = 0
+        let nonELLRecommendationsCount = 0
+
+        DIAGNOSTIC_LEVELS.forEach((diagnosticLevel) => {
+          if (ellRecommendations[diagnosticLevel].student_ids.length) {
+            ellRecommendationsCount += ellRecommendations[diagnosticLevel].student_ids.length
+            const recommendedStudentNames = ellLearningSequences.filter((studentLearningSequence) => ellRecommendations[diagnosticLevel].student_ids.includes(studentLearningSequence.user_id)).map(studentLearningSequence => studentLearningSequence.user_name)
+            newRecommendedDiagnosticsByLevel[`ell_${diagnosticLevel}`].push({ id: classroomId, name: classroom.classroom_name, studentNames: recommendedStudentNames})
+          }
+
+          if (nonELLRecommendations[diagnosticLevel].student_ids.length) {
+            nonELLRecommendationsCount += nonELLRecommendations[diagnosticLevel].student_ids.length
+            const recommendedStudentNames = nonELLLearningSequences.filter((studentLearningSequence) => nonELLRecommendations[diagnosticLevel].student_ids.includes(studentLearningSequence.user_id)).map(studentLearningSequence => studentLearningSequence.user_name)
```

Comment:
These lines should be refactored to conform to the 120 character-width standard. This particular line is 209 characters long. The line wrapping makes the code difficult to read.